pkg/build/controller/buildimagechangetrigger/buildimagechangetrigger_controller.go

```diff
@@ -0,0 +1,287 @@
+				continue
+			}
+
+			latestReference, ok = imageapi.ResolveLatestTaggedImage(stream, tag)
+			if !ok {
+				glog.V(4).Infof("unable to find tagged image: no image recorded for %s/%s:%s", stream.Namespace, stream.Name, tag)
+				continue
+			}
+			glog.V(4).Infof("Found ImageStream %s/%s with tag %s", stream.Namespace, stream.Name, tag)
+
+			// (must be different) to trigger a build
+			last := trigger.ImageChange.LastTriggeredImageID
+			next := latestReference
+
+			if len(last) == 0 || (len(next) > 0 && next != last) {
+				triggeredImage = next
+				shouldBuild = true
+				// it doesn't really make sense to have multiple image change triggers any more,
+				// so just exit the loop now
+				break
+			}
+		}
+
+		if shouldBuild {
+			glog.V(4).Infof("Running build for BuildConfig %s/%s", config.Namespace, config.Name)
+			buildCauses := []buildapi.BuildTriggerCause{}
+			// instantiate new build
+			request := &buildapi.BuildRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      config.Name,
+					Namespace: config.Namespace,
+				},
+				TriggeredBy: append(buildCauses,
+					buildapi.BuildTriggerCause{
+						Message: buildapi.BuildTriggerCauseImageMsg,
+						ImageChangeBuild: &buildapi.ImageChangeCause{
+							ImageID: latestReference,
+							FromRef: from,
+						},
+					},
+				),
+				TriggeredByImage: &kapi.ObjectReference{
+					Kind: "DockerImage",
+					Name: triggeredImage,
+				},
+				From: from,
+			}
+			if _, err := bc.bcInstantiator.Instantiate(config.Namespace, request); err != nil {
+				if kerrors.IsConflict(err) {
+					utilruntime.HandleError(fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", config.Namespace, config.Name, err))
+				} else {
+					utilruntime.HandleError(fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %#v", config.Namespace, config.Name, err))
+				}
+				hasError = true
+				continue
+			}
+		}
+	}
+	if hasError {
+		return fmt.Errorf("an error occurred processing 1 or more build configurations; the image change trigger for image stream/namespace %s/%s and docker image repository %s will be retried", stream.Name, stream.Namespace, stream.Status.DockerImageRepository)
+	}
+
+	return nil
+}
+
+func (bc *BuildImageChangeTriggerController) handleError(err error, key interface{}, stream *imageapi.ImageStream) {
+	if err == nil {
+		bc.queue.Forget(key)
+		return
+	}
+
+	if strategy.IsFatal(err) {
+		glog.V(2).Infof("Will not retry fatal error for imagestream/namespace %s/%s: %v", stream.Namespace, stream.Name, err)
+		bc.queue.Forget(key)
+		return
+	}
+
+	if bc.queue.NumRequeues(key) < maxRetries {
+		glog.V(4).Infof("Retrying imagestream/namespace %s/%s: %v", stream.Namespace, stream.Name, err)
+		bc.queue.AddRateLimited(key)
+		return
+	}
+
+	glog.V(2).Infof("Giving up retrying imagestream/namespace %s/%s: %v", stream.Namespace, stream.Name, err)
```

Comment:
same formatting changes for all 3 of these logs:  namespace/name  (actually your formatting is correct, but the "imagestream/namespace" text is reversed.  I'd just change it to "giving up retrying imagestream %s/%s", and similar for the 2 above)